src/functions/fetchRequests/fetchRequestsUser.js

```diff
@@ -0,0 +1,257 @@
+// :: users - holds the users 
+
+// id - integer (auto create by MySql/Postgres), PK
+// username - String, unique, required
+// password - String, unique, required
+// fullName - String, required
+// email    - String, required, unique
+// bio      - Text (String)
+// hasRopes - Boolean, required, ??
+// location - String, required, ??
+
+// ------------------------------------------------------
+
+// POST /user/login {username, password} => {userObj, token}   :: login a user by validating his username and password
+
+// POST /user {userObj} => {userObj, token}     :: create a user in the database
+// PATCH /user {userObj, token} => {userObj}    :: update a user in the database
+// DELETE /user {token} => {success/failure}    :: delete a user in the database
+
+// GET /user/self {token} => {userObj}          :: get logged-in user's information
+// GET /user/:username => {userObj}             :: get another user's data - maybe a potential buddy or a buddy
+// GET /user/friends {token} => [{userObj}, ]   :: get list with the user's buddies
+
+// ------------------------------------------------------
+
+// loginUser (userObj, setUser)         :: loginUser( {userObj}, setUser ) => fetch(); use setUser( loggedUser ); save token
+// logoutUser (setUser)                 :: logoutUser( token ) => remove token; setUser( {} )
+
+// createUser (userObj, setUser)        :: createUser( {userObj}, setUser ) => fetch(); use setUser( newUser ); save token
+// updateUser (userObj, setUser)        :: updateUser( {userObj}, setUser, token ) => fetch(); use setUser( user ); 
+// deleteUser (userObj, setUser)        :: deleteUser( {userObj,password}, token ) => fetch(); setUser( {} ); remove token;
+
+// retrieveUser (setUser)               :: retrieveUser( token ) => fetch(token); retrieve userObj
+// retrieveUserByUsername ( username, setUser) :: retrieveUserByUsername( username ) => fetch(token); retrieve userObj
+// retrieveUserFriends (setFriendList)  :: retrieveUserFriends( setFriends ) => fetch(token); retrieve [userObj,]
+
+// ------------------------------------------------------
+
+
+// createUser( {userObj}, setUser ) => fetch(); use setUser( newUser ); save token
+export const createUser = async (userObj, setUser) => {
+
+    try {
+        // POST /user {userObj} => {userObj, token}
+        const res = await fetch(`${process.env.REACT_APP_REST_API}/user`, {
+            method: "POST",
+            headers: { "Content-Type": "application/json" },
+            body: JSON.stringify( userObj ),
+        });
+
+
+        const data = await res.json();
+        console.log('-> createUser(), data:', data);
+
+        if (!res.ok || data.error) throw new Error(`FetchError ${res.status} : ${data.error || data}`)
+
+        // since Michael doesn't want the token in localStorage, attach it to userObj
```

Comment:
This is fine for now. I'm going to look into cookies for the token storage. If you have a look into them them please let me know.